cirq/circuits/circuit.py

```diff
@@ -776,13 +776,13 @@ def are_all_matches_terminal(self, predicate: Callable[['cirq.Operation'], bool]
         if not all(
             self.next_moment_operating_on(op.qubits, i + 1) is None
             for (i, op) in self.findall_operations(predicate)
-            if _get_op_circuit(op) is None
+            if getattr(op.untagged, 'circuit', None) is None
```

Comment:
I think it would more clearly convey intent here to check the type instead
```suggestion
            if not isinstance(op.untagged, CircuitOperation)
```
Would have to import `CircuitOperation` in `are_all_matches_terminal` and `are_any_matches_terminal`, but that should be fine. WDYT?